util/src/main/java/io/grpc/util/SpiffeIdParser.java

```diff
@@ -0,0 +1,10 @@
+package io.grpc.util;
+
+public interface SpiffeIdParser {
```

Comment:
I'm a bit confused why the "parser" would be the object rather than the "SPIFFE ID" being the object. Having both adds an unnecessary indirection layer, I think. For example, I think it would be simpler for users of this interface to have

```
class SpiffeId {
  static SpiffeId parse(String uri);
  String getTrustDomain();
  String getPath();
};
```

WDYT?

Reply:
Personally I prefer more functional style - "here is the input, give me the idempotent output"

After looking at the package Eric mentioned (`io.grpc.internal`), the closest example is [JsonParser](https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/JsonParser.java) so I'd still prefer the extra layer I currently have.

Reply:
Per our offline discussion let's change SpiffeIdInfo to SpiffeId